prover/src/constraints/evaluator/logup_gkr.rs

```diff
@@ -293,3 +335,23 @@ impl<E: FieldElement> TransitionDivisorEvaluator<E> {
             - E::BaseField::ONE
     }
 }
+
+/// Computes the evaluations of the s-column divisor.
+///
+/// The divisor for the s-column is $X^n - 1$ where $n$ is the trace length. This means that
+/// we need only compute `ce_blowup` many values and thus only that many exponentiations.
+fn compute_s_col_divisor<E: FieldElement>(
+    ce_domain_size: usize,
+    domain: &StarkDomain<E::BaseField>,
+    trace_length: usize,
+) -> Vec<E::BaseField> {
+    let degree = trace_length as u32;
+    let mut result = Vec::with_capacity(ce_domain_size);
+
+    for row in 0..domain.trace_to_ce_blowup() {
+        let x = domain.get_ce_x_at(row).exp(degree.into()) - E::BaseField::ONE;
```

Comment:
I am not sure this is the case. As mentioned in the comment at the top of the function, this will call the exponentiation procedure only `domain.trace_to_ce_blowup()` and this is negligible for all practical purposes.

Reply:
Ah! Sorry I missed that.